src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp

```diff
@@ -772,11 +777,12 @@ void DiskObjectStorageTransaction::createFile(const std::string & path)
 
 void DiskObjectStorageTransaction::copyFile(const std::string & from_file_path, const std::string & to_file_path, const WriteSettings & settings)
 {
-    /// NOTE: For native copy we can ignore throttling, so no need to use WriteSettings
-    UNUSED(settings);
+    ReadSettings read_settings = Context::getGlobalContextInstance()->getReadSettings();
+    read_settings.local_throttler = settings.local_throttler;
+    read_settings.remote_throttler = settings.remote_throttler;
```

Comment:
I don't quite understand this change. You take throttlers from the write settings and put them into the read settings. It seems we didn't do that before.

Reply:
Yes. This is because copy accept read settings, but all it needs is actually throttling.

Reply:
@vitlibar so, do you have any suggestions here?
I thought about adding write throttler for copy as well, but decided not to complicate things.

Reply:
It seems too hacky. It looks like it would be better to add `ReadSettings` to the arguments of `copyFile()` here and in the interfaces of `IDisk` and `IObjectStorage`.

Reply:
OK